core/services/relay/evm/mercury/v1/reportcodec/report_codec.go

```diff
@@ -115,49 +100,25 @@ func (r *ReportCodec) MaxReportLength(n int) (int, error) {
 }
 
 func (r *ReportCodec) CurrentBlockNumFromReport(report ocrtypes.Report) (int64, error) {
-	reportElems := map[string]interface{}{}
-	if err := ReportTypes.UnpackIntoMap(reportElems, report); err != nil {
-		return 0, errors.Errorf("error during unpack: %v", err)
-	}
-
-	blockNumIface, ok := reportElems["currentBlockNum"]
-	if !ok {
-		return 0, errors.Errorf("unpacked report has no 'currentBlockNum' field")
-	}
-
-	blockNum, ok := blockNumIface.(uint64)
-	if !ok {
-		return 0, errors.Errorf("cannot cast blockNum to int64, type is %T", blockNumIface)
+	decoded, err := r.Decode(report)
+	if err != nil {
+		return 0, err
 	}
-
-	if blockNum > math.MaxInt64 {
-		return 0, errors.Errorf("blockNum overflows max int64, got: %d", blockNum)
+	if decoded.CurrentBlockNum > math.MaxInt64 {
+		return 0, fmt.Errorf("CurrentBlockNum=%d overflows max int64", decoded.CurrentBlockNum)
 	}
-
-	return int64(blockNum), nil
+	return int64(decoded.CurrentBlockNum), nil
```

Comment:
shouldn't this be uint64?

Reply:
block number is generally int64 everywhere because this is what go-ethereum uses internally. It is also what mercury server returns to us. We have validations to ensure it can't be negative.